test/native/perfEventsLinuxTest.cpp

```diff
@@ -0,0 +1,269 @@
+
+#define ASSERT_PROBE(event_type, name_, type_, default_interval_, config_, probeName, config2_, counter_arg_) \
+  ASSERT_NE(event_type, NULL);                                                                                \
+  CHECK_EQ(event_type->name, name_);                                                                          \
+  CHECK_NE(type_, 0);                                                                                         \
+  CHECK_EQ(event_type->default_interval, default_interval_);                                                  \
+  CHECK_EQ(event_type->config, config_);                                                                      \
+  CHECK_EQ(((const char*)event_type->config1), probeName);                                                    \
+  CHECK_EQ(event_type->config2, config2_);                                                                    \
+  CHECK_EQ(event_type->counter_arg, counter_arg_)
+
+TEST_CASE(ForName_Predefined_cpu) {
+  PerfEventType* event_type = PerfEventType::forName("cpu");
+  ASSERT_EVENT_TYPE(event_type, "cpu-clock", PERF_TYPE_SOFTWARE, DEFAULT_INTERVAL, PERF_COUNT_SW_CPU_CLOCK, 0, 0, 0);
+}
+
+TEST_CASE(ForName_Predefined_cpu_clock) {
+  PerfEventType* event_type = PerfEventType::forName("cpu-clock");
+  ASSERT_EVENT_TYPE(event_type, "cpu-clock", PERF_TYPE_SOFTWARE, DEFAULT_INTERVAL, PERF_COUNT_SW_CPU_CLOCK, 0, 0, 0);
+}
+
+TEST_CASE(ForName_Invalid_space) {
+  PerfEventType* event_type = PerfEventType::forName(" ");
+  ASSERT_EQ(event_type, NULL);
+}
+
+TEST_CASE(ForName_Breakpoint_Invalid_Zero) {
+  PerfEventType* event_type = PerfEventType::forName("mem:0");
+  ASSERT_EQ(event_type, NULL);
+}
+
+TEST_CASE(ForName_Breakpoint_Invalid) {
+  PerfEventType* event_type = PerfEventType::forName("mem:foo_unknownSymbol");
+  ASSERT_EQ(event_type, NULL);
+}
+
+TEST_CASE(ForName_Breakpoint_InvalidColon2) {
+  PerfEventType* event_type = PerfEventType::forName("mem::");
+  ASSERT_EQ(event_type, NULL);
+}
+
+TEST_CASE(ForName_Breakpoint_InvalidColon) {
+  PerfEventType* event_type = PerfEventType::forName("mem:");
+  ASSERT_EQ(event_type, NULL);
+}
+
+TEST_CASE(ForName_Breakpoint_HexAddr) {
+  PerfEventType* event_type = PerfEventType::forName("mem:0x123");
+  ASSERT_BP(event_type, HW_BREAKPOINT_RW, 0x123, 1, 0);
+}
+
+TEST_CASE(ForName_Breakpoint_DecAddr) {
+  PerfEventType* event_type = PerfEventType::forName("mem:123");
+  ASSERT_EQ(event_type, NULL);
+}
+
+TEST_CASE(ForName_Breakpoint_Addr_DecOffset) {
+  PerfEventType* event_type = PerfEventType::forName("mem:0x123+22000");
+  ASSERT_BP(event_type, HW_BREAKPOINT_RW, 22291, 1, 0);
+}
+
+TEST_CASE(ForName_Breakpoint_Addr_HexOffset) {
+  PerfEventType* event_type = PerfEventType::forName("mem:0x123+0x22000");
+  ASSERT_BP(event_type, HW_BREAKPOINT_RW, 0x22123, 1, 0);
+}
+
+TEST_CASE(ForName_Breakpoint_Addr_HexOffset_DecLen) {
+  PerfEventType* event_type = PerfEventType::forName("mem:0x123+0x22000/16");
```

Comment:
To be pedantic, correct breakpoint length can be only 1, 2, 4 or 8.
It's OK for the parser to accept other values though - they will be rejected later by perf_event_open API. A test should verify either 100% correct value or 100% incorrect value; for other values, behavior is implementation defined and therefore should not occur in assertions.